### PR TITLE
Date nanos rollup fix

### PR DESCRIPTION
**Pull request by petardz**
*Issue #, if available:* #767

*Description of changes:*

Added trimming of nanos part of "epoch_millis" timestamp when date_histogram type used is date_nanos

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/indexmanagement/rollup/RollupMetadataService.kt [hunk 1]
```diff
@@ -194,10 +194,15 @@ class RollupMetadataService(val client: Client, val xContentRegistry: NamedXCont
 
             // Get the doc value field of the dateHistogram.sourceField for the first search hit converted to epoch millis
             // If the doc value is null or empty it will be treated the same as empty doc hits
-            val firstHitTimestamp = response.hits.hits.first().field(dateHistogram.sourceField).getValue<String>()?.toLong()
-                ?: return StartingTimeResult.NoDocumentsFound
-
-            return StartingTimeResult.Success(getRoundedTime(firstHitTimestamp, dateHistogram))
+            var firstHitTimestampAsString = response.hits.hits.first().field(dateHistogram.sourceField).getValue<String>()
```
Comment (bowenlan-amzn) [inline src/main/kotlin/org/opensearch/indexmanagement/rollup/RollupMetadataService.kt:197]: `epoch_millis` format seems not working as expected
Tried `strict_date_optional_time` format and it seems to work and return a millisecond representation of a nanosecond field, can we use this format instead then.

File: src/test/kotlin/org/opensearch/indexmanagement/rollup/runner/RollupRunnerIT.kt [hunk 2]
```diff
@@ -1253,6 +1254,71 @@ class RollupRunnerIT : RollupRestTestCase() {
...
+        )
+
+        indexDoc(index, "1", """{"purchaseDate": 1683149130000.6497, "itemName": "shoes", "itemPrice": 100.5}""".trimIndent())
+        indexDoc(index, "2", """{"purchaseDate": 1683494790000, "itemName": "shoes", "itemPrice": 30.0}""".trimIndent())
+        indexDoc(index, "3", """{"purchaseDate": "2023-05-08T18:57:33.743656789Z", "itemName": "shoes", "itemPrice": 60.592}""".trimIndent())
+
+        refreshAllIndices()
+
+        val job = Rollup(
+            id = "rollup_with_alias_99243411",
```
Comment (bowenlan-amzn) [inline src/test/kotlin/org/opensearch/indexmanagement/rollup/runner/RollupRunnerIT.kt:1283]: Don't use the same id as other test

Comment (petardz) [reply src/test/kotlin/org/opensearch/indexmanagement/rollup/runner/RollupRunnerIT.kt]: oops.. changed it!

**Review by bowenlan-amzn — APPROVED** [review]
Thanks for the change!

**Pull request merged**